pyproject.toml

```diff
@@ -67,9 +67,13 @@ multi_line_output = 3
 ignore_whitespace = true
 
 [[tool.mypy.overrides]]
-module = ["brownie.*", "twisted.*"]
+module = ["ape.*", "twisted.*"]
 ignore_missing_imports = true
 
+[[tool.mypy.overrides]]
+module = 'beamer.tests.*'
+disable_error_code = "attr-defined,name-defined"
+
```

Comment:
Why do we need this?

Reply:
if i recall correctly because of ape.reverts part for example, reverts is not a part of ape i guess also it is true for project.Contract i guess.

Reply:
Yeah, most of the errors are of that sort:
```
beamer/tests/contracts/test_request_manager.py:197: error: Module has no attribute "reverts"  [attr-defined]
```

I think we definitely need to address this some way, instead of just getting rid of the error in tests like that.

There is also 
```
beamer/tests/contracts/test_request_manager.py:655: error: "ProviderAPI" has no attribute "web3"  [attr-defined]
```
as well as
```
beamer/tests/agent/unit/test_handlers.py:92: error: "Callable[[Optional[TxParams], Union[Literal['latest', 'earliest', 'pending', 'safe', 'finalized'], BlockNumber, Hash32, HexStr, HexBytes, int], Optional[Dict[ChecksumAddress, CallOverrideParams]], Opti
onal[bool]], Any]" has no attribute "return_value"  [attr-defined]
beamer/tests/agent/unit/test_handlers.py:100: error: "Callable[[Optional[TxParams]], HexBytes]" has no attribute "return_value"  [attr-defined]
beamer/tests/agent/unit/test_handlers.py:101: error: "ContractFunction" has no attribute "web3"  [attr-defined]
beamer/tests/agent/unit/test_handlers.py:112: error: "Callable[[Optional[TxParams], Union[Literal['latest', 'earliest', 'pending', 'safe', 'finalized'], BlockNumber, Hash32, HexStr, HexBytes, int], Optional[Dict[ChecksumAddress, CallOverrideParams]], Opt
ional[bool]], Any]" has no attribute "return_value"  [attr-defined]
beamer/tests/agent/unit/test_handlers.py:122: error: "ContractFunction" has no attribute "web3"  [attr-defined]
beamer/tests/conftest.py:28: error: Name "project.Resolver" is not defined  [name-defined]
beamer/tests/conftest.py:29: error: Name "project.FillManager" is not defined  [name-defined]
beamer/tests/conftest.py:30: error: Name "project.RequestManager" is not defined  [name-defined]
beamer/tests/conftest.py:31: error: Name "project.TestL1Messenger" is not defined  [name-defined]
beamer/tests/conftest.py:32: error: Name "project.TestL2Messenger" is not defined  [name-defined]
```

I think we need to think about those cases and handle them.

Reply:
you mean line by line going there and adding # type: ignore ?

Reply:
> you mean line by line going there and adding # type: ignore ?

For sure not. :-)
Need to find a saner way to deal with those.

Reply:
still not sure about this, for example looked at brownie for reverts, i couldnt find any difference in regards with ape and brownie has no problem with reverts but ape does.